tools/tplgtool.py

```diff
@@ -854,11 +854,15 @@ def get_tplg_paths(cmd_args):
             tplg_paths.append(f)
         return tplg_paths
 
-    def connect(graph, raw_graph_list, nodes_added):
+    def connect(graph, raw_graph_list, nodes_added, connnected_link):
         for raw_graph in raw_graph_list:
             for link in raw_graph:
                 if link[0] in nodes_added and link[2] in nodes_added:
-                    graph.edge(link[0], link[2])
+                    link_str = link[0] + '->' + link[2]
+                    if link_str not in connnected_link:
+                        graph.edge(link[0], link[2])
+                        connnected_link.append(link_str)
+
```

Comment:
You don't need to use strings, they use more CPU and memory. This should work:
```python
new_edge = (link[0], link[2])
if new_edge not in edges:
    graph.edge(*new_edge)
    edges.append(new_edge)
```
PS: stick to the graphviz terminology, this avoids wondering whether it's something different.